File: frontend/src/app/models/EnterRoom.ts
import {BaseTransferObject} from "./baseTransferObject";
import {Message} from "./entities";

export class ClientWantsToEnterRoom extends BaseTransferObject {
  roomId?: number;

  constructor(init?: Partial<ClientWantsToEnterRoom>) {
    super();
    Object.assign(this, init);
  }

}

export class ServerLetsClientEnterRoom extends BaseTransferObject {
  roomId?: number;
  recentMessages?: Message[];
  constructor(init?: Partial<ServerLetsClientEnterRoom>) {
    super();
    Object.assign(this, init);
  }

}

File: backend/shared/Models/WebsocketTransferObjects/EnterRoom.cs
using System.ComponentModel.DataAnnotations;

namespace core.Models.WebsocketTransferObjects;

public class ClientWantsToEnterRoom : BaseTransferObject
{

    [Required, Range(1, int.MaxValue)]
    public int roomId { get; set; }
}

public class ServerLetsClientEnterRoom : BaseTransferObject
{

    public int roomId { get; set; }
    public IEnumerable<Message>? recentMessages { get; set; }
}

